### PR TITLE
feat add IT 1 to 3

### PR DESCRIPTION
**Pull request by Eagerlearn**
Craig Cardwell

### CODE REVIEW

File: lib/dealership.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      car.total_cost
+    end
+  end
+
+  def details
+    dealer_details = {"total_value" => total_value, "address" => @address}
+  end
+
+  def average_price_of_car
+    (total_value / inventory_count).to_s
```
Comment (kat3kasper) [inline lib/dealership.rb:40]: Really close here! Was hoping that you were able to add some formatting to include the "," when working with larger numbers. You can read about some examples here: https://rick-moore.medium.com/formatting-number-strings-in-ruby-4da35d5282e3

File: lib/dealership.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  def cars_by_make(make)
+    @inventory.find_all do |car|
+      car.make == make
+    end
+  end
+
+  def total_value
+    @inventory.sum do |car|
+      car.total_cost
+    end
```
Comment (kat3kasper) [inline lib/dealership.rb:32]: Very clean! I like this

File: spec/car_spec.rb [hunk 3]
```diff
@@ -37,4 +37,16 @@
 
     expect(car.total_cost).to eq(54000)
   end
+
+  it 'has no color by default' do
```
Comment (kat3kasper) [inline spec/car_spec.rb:41]: I love the sad path test here!

File: spec/dealership_spec.rb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    car_3 = Car.new("Toyota Tercel", 500, 48)
+    car_4 = Car.new("Chevrolet Bronco", 1250, 24)
+    dealership.add_car(car_1)
+    dealership.add_car(car_2)
+    dealership.add_car(car_3)
+    dealership.add_car(car_4)
+    expect(dealership.details).to eq({"total_value" => 156000, "address" => "123 Main Street"})
+  end
+
+  it 'total_value' do
```
Comment (kat3kasper) [inline spec/dealership_spec.rb:79]: The name of this test doesn't really match what you seem to be testing for which is the `average_price_of_car`

**Review by kat3kasper — COMMENTED** [review]
Nice job with getting through 3 iterations! Love how clean and easy to read the object classes were.

**Comment by Eagerlearn** [comment]
Thanks for the feedback and the resource link! I'll be taking a look at
it.  I gotta also remember to be more aware of copy and paste and ensure my
tests are clean and aligned with the interaction pattern. 👍

**Pull request closed without merging**